modin/pandas/indexing.py

```diff
@@ -764,32 +764,94 @@ def _loc(df):
                 new_query_compiler=self.df._default_to_pandas(_loc)._query_compiler
             )
             return
-        row_loc, col_loc, _ = self._parse_row_and_column_locators(key)
-        if isinstance(row_loc, list) and len(row_loc) == 1:
-            if row_loc[0] not in self.qc.index:
-                index = self.qc.index.insert(len(self.qc.index), row_loc[0])
-                self.qc = self.qc.reindex(labels=index, axis=0)
-                self.df._update_inplace(new_query_compiler=self.qc)
+        row_loc, col_loc, ndims = self._parse_row_and_column_locators(key)
+        append_axis = self._check_missing_loc(row_loc, col_loc)
+        if ndims >= 1 and append_axis is not None:
+            # We enter this codepath if we're either appending a row or a column
+            if append_axis:  # Appending new column
+                if is_scalar(col_loc) or len(col_loc) == 1:
+                    if is_scalar(item):
+                        item = [item] * len(self.qc.index)
+                    if not isinstance(item, (Series, DataFrame)):
+                        item = Series(item, name=col_loc[0])
+                    elif isinstance(item, Series):
+                        item = item.rename(index=col_loc[0])
+                    else:
+                        item = item.rename(columns=lambda x: col_loc[0])
```

Comment:
this bit feels weird, it would give `col_loc[0]` name to _all_ columns in there, won't it?

Reply:
We check if `col_loc` is a length 1 array a few lines above this using `if is_scalar(col_loc) or len(col_loc) == 1:`. The `item` passed in should only be one column as well.